### PR TITLE
(SIMP-6296) Multiple fixes

### PR DESCRIPTION
**Pull request by trevor-vaughan**
- Ensure that the acceptance tests pass and bring up the bridge device
- Fix the fact that network::global could break networking
- Expose failures during Exec's of network-related activity
- Add NetworkManager support

SIMP-4891 #close
SIMP-5665 #close
SIMP-6296 #close
SIMP-2753 #close

### CODE REVIEW

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-network/pipelines/53550141

File: manifests/eth.pp [hunk 1]
```diff
@@ -93,6 +93,7 @@
 # @param net_type
 # @param netmask
 # @param network
+# @param nm_controlled
```
Comment (lnemsick-simp) [inline manifests/eth.pp]: This is a new feature and warrants a minor version bump

File: manifests/eth.pp [hunk 2]
```diff
@@ -144,51 +145,55 @@
...
-  Optional[Integer]                 $mtu                              = undef,
-  Optional[String]                  $net_type                         = undef,
+  Optional[String[1]]               $master                           = undef,
+  Optional[String[1]]               $metric                           = undef,
+  Optional[Integer[1]]              $mtu                              = undef,
+  Optional[String[1]]               $net_type                         = undef,
   Optional[Simplib::IP]             $netmask                          = undef,
   Optional[Simplib::IP]             $network                          = undef,
-  Optional[String]                  $nozeroconf                       = undef,
+  Boolean                           $nm_controlled                    = pick(fact('simplib_networkmanager.enabled'), false),
```
Comment (lnemsick-simp) [inline manifests/eth.pp]: Minimum version of simplib in metadata.json should be updated to the version of simplib that provides this fact

**Comment by jeannegreulich** [comment]
While you are looking at everyones comments can you change simpcat to concat in the network module?

File: manifests/eth.pp (hunk 1 above)
Comment (sharkbruhaha) [reply manifests/eth.pp]: :+1:

File: manifests/eth.pp (hunk 2 above)
Comment (sharkbruhaha) [reply manifests/eth.pp]: :+1:

Comment (jeannegreulich) [reply manifests/eth.pp]: fixed

**Comment by sharkbruhaha** [comment]
> 
> 
> heavy_check_mark https://gitlab.com/simp/pupmod-simp-network/pipelines/53550141

Can you add acceptance tests for pup6? (and go ahead and remove pup4)

**Comment by trevor-vaughan** [comment]
@jeannegreulich Having `bootproto=dhcp` *should* override all of the other settings, but I'll go take a look again.

**Comment by sharkbruhaha** [comment]
The 5.5.7 acceptance test keeps failing (have prodded a couple times now): https://gitlab.com/simp/pupmod-simp-network/pipelines/56524412

**Comment by trevor-vaughan** [comment]
@jeannegreulich Good call. I've just switched things over to use `ifup/ifdown` and/or `nmcli` depending on the file target.

I'm also splitting out the network/networkmanager service since technically both could be used at the same time.

**Comment by trevor-vaughan** [comment]
Rebased and updated

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-network/pipelines/57317763

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-network/pipelines/57318024

**Comment by trevor-vaughan** [comment]
The GitLab tests are failing but it looks like it might be runner issues. I can't get it to fail locally.

**Pull request merged**